src/HucaresServer.Storage.UnitTests/DetectedPlateHelperTests.cs

```diff
@@ -0,0 +1,122 @@
+                expectedCamId, expectedImgUrl, expectedConfidence);
+
+            //Assert
+            A.CallTo(() => fakeDbContextFactory.BuildHucaresContext())
+                .MustHaveHappenedOnceExactly();
+
+            A.CallTo(() => fakeDbSet.Add(result))
+                .MustHaveHappened();
+
+            A.CallTo(() => fakeHucaresContext.SaveChanges())
+                .MustHaveHappenedOnceExactly();
+            
+            result.PlateNumber.ShouldBe(expectedPlateNumber);
+            result.DetectedDateTime.ShouldBe(expectedDetectedDateTime);
+            result.CamId.ShouldBe(expectedCamId);
+            result.ImgUrl.ShouldBe(expectedImgUrl);
+            result.Confidence.ShouldBe(expectedConfidence);
+
+        }
+        
+        [TestMethod]
+        public void InsertNewDetectedPlate_WhenConfidenceNotCorrect_ShouldThrowError()
+        {
+
+            //Arrange
+            var fakeIQueryable = new List<DetectedLicensePlate>().AsQueryable();
+            var fakeDbSet = A.Fake<DbSet<DetectedLicensePlate>>();
+
+            var fakeHucaresContext = A.Fake<HucaresContext>();
+            A.CallTo(() => fakeHucaresContext.DetectedLicensePlates)
+                .Returns(fakeDbSet);
+
+            var fakeDbContextFactory = A.Fake<IDbContextFactory>();
+            A.CallTo(() => fakeDbContextFactory.BuildHucaresContext())
+                .Returns(fakeHucaresContext);
+
+            var detectedPlateHelper = new DetectedPlateHelper(fakeDbContextFactory);
+
+            //Act and Assert
+            Assert.ThrowsException<ArgumentException>(() => detectedPlateHelper.InsertNewDetectedPlate(
+                "ABC001", new DateTime(2018, 09, 29),
+                1, "http://localhost:6969/images/cam01_21080929_235959", 1.1));
+            
+            A.CallTo(() => fakeDbContextFactory.BuildHucaresContext())
+                .MustNotHaveHappened();
+
+        }
+        
+        [TestMethod]
+        public void InsertNewDetectedPlate_WhenUriNotCorrect_ShouldThrowError()
+        {
+
+            //Arrange
+            var fakeIQueryable = new List<DetectedLicensePlate>().AsQueryable();
+            var fakeDbSet = A.Fake<DbSet<DetectedLicensePlate>>();
+
+            var fakeHucaresContext = A.Fake<HucaresContext>();
+            A.CallTo(() => fakeHucaresContext.DetectedLicensePlates)
+                .Returns(fakeDbSet);
+
+            var fakeDbContextFactory = A.Fake<IDbContextFactory>();
+            A.CallTo(() => fakeDbContextFactory.BuildHucaresContext())
+                .Returns(fakeHucaresContext);
+
+            var detectedPlateHelper = new DetectedPlateHelper(fakeDbContextFactory);
+
+            //Act and Assert
+            Assert.ThrowsException<UriFormatException>(() => detectedPlateHelper.InsertNewDetectedPlate(
+                "ABC001", new DateTime(2018, 09, 29),
+                1, "We choose to go to the moon in this decade and do the other things, not because they are easy, " +
```

Comment:
Beautiful writing but not suited for master branch. Change to something shorter, which would also tell what its purpose, for example: "notAndUri" or something of the like.